dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -22,13 +22,33 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 {
     PushNotificationChannel::PushNotificationChannel(winrt::Windows::PushNotificationChannel const& channel): m_channel(channel) {}
 
+    PushNotificationChannel::PushNotificationChannel(hstring const& channelUri, winrt::Windows::Foundation::DateTime const& channelExpirationTime)
+    {
+        m_channelUri = winrt::Windows::Uri{ channelUri };
+        m_channelExpirationTime = channelExpirationTime;
+    }
+
     winrt::Windows::Uri PushNotificationChannel::Uri()
     {
-        return winrt::Windows::Uri{ m_channel.Uri() };
+        if (m_channel)
+        {
+            return winrt::Windows::Uri{ m_channel.Uri() };
+        }
+        else
+        {
+            return m_channelUri;
+        }
     }
     winrt::Windows::DateTime PushNotificationChannel::ExpirationTime()
     {
-        return m_channel.ExpirationTime();
+        if (m_channel)
+        {
+            return m_channel.ExpirationTime();
+        }
+        else
+        {
+            return m_channelExpirationTime;
+        }
     }
     void PushNotificationChannel::Close()
     {
```

Comment:
Calling this function will end up in a crash, when the object has been constructed using the new constructor. In which case m_channel is initialized to nullptr.